File: app/src/main/java/com/durbinlabs/roomdemo/adapters/RecyclerViewAdapter.java
package com.durbinlabs.roomdemo.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.durbinlabs.roomdemo.R;
import com.durbinlabs.roomdemo.database.AppDatabase;
import com.durbinlabs.roomdemo.model.Book;
import com.durbinlabs.roomdemo.model.Client;
import com.durbinlabs.roomdemo.model.ClientDataModel;

import java.util.List;

/**
 * Created by hp on 12/27/2017.
 */

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {
    private static final String TAG = RecyclerViewAdapter.class.getSimpleName();
    private List<ClientDataModel> modelList;
    private Context context;
    private AppDatabase db;
    private View.OnLongClickListener listener;

    public RecyclerViewAdapter(List<ClientDataModel> modelList, Context context, View.OnLongClickListener
            listener) {
        this.modelList = modelList;
        this.context = context;
        db = AppDatabase.getInstance(context);
        this.listener = listener;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.recycler_row, null);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, final int position) {
        int pos = holder.getAdapterPosition();
        ClientDataModel model = modelList.get(pos);
        Book book = new Book(model.getTotalBook());
        Client client = new Client(model.getId(), model.getName(), model.getAge(), book);

        holder.tvId.setText("ID: " + client.getId());
        holder.tvName.setText("Name: " + client.getName());
        holder.tvAge.setText("Age: " + client.getAge());
        holder.tvBookNo.setText(context.getString(R.string.total_book_) + " " + client.getBook().getTotalBook() + "");


        holder.itemView.setTag(client);
        holder.itemView.setOnLongClickListener(listener);
    }


    @Override
    public int getItemCount() {
        return modelList.size();
    }


    protected class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView tvName, tvAge, tvBookNo, tvId;

        public MyViewHolder(View itemView) {
            super(itemView);

            tvId = itemView.findViewById(R.id.tvID);
            tvName = itemView.findViewById(R.id.tvName);
            tvAge = itemView.findViewById(R.id.tvAge);
            tvBookNo = itemView.findViewById(R.id.tvBookNo);
        }
    }

    public void addDataToDataModel(List<ClientDataModel> models) {
        this.modelList = models;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/durbinlabs/roomdemo/adapters/RecyclerViewAdapter.kt
package com.durbinlabs.roomdemo.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import com.durbinlabs.roomdemo.R
import com.durbinlabs.roomdemo.database.AppDatabase
import com.durbinlabs.roomdemo.model.Book
import com.durbinlabs.roomdemo.model.Client
import com.durbinlabs.roomdemo.model.ClientDataModel

/**
 * Created by hp on 12/27/2017.
 */

class RecyclerViewAdapter(
    private var modelList: List<ClientDataModel>?,
    private val context: Context,
    private val listener: View.OnLongClickListener
) : RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(context).inflate(R.layout.recycler_row, null)
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val pos = holder.adapterPosition
        val model = modelList!![pos]
        val book = Book(model.totalBook)
        val client = Client(model.id, model.name!!, model.age, book)

        holder.tvId.text = "ID: " + client.id
        holder.tvName.text = "Name: " + client.name!!
        holder.tvAge.text = "Age: " + client.age
        holder.tvBookNo.text =
            context.getString(R.string.total_book_) + " " + client.book!!.totalBook + ""


        holder.itemView.tag = client
        holder.itemView.setOnLongClickListener(listener)
    }


    override fun getItemCount(): Int {
        return modelList!!.size
    }


    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvName: TextView
        var tvAge: TextView
        var tvBookNo: TextView
        var tvId: TextView

        init {

            tvId = itemView.findViewById(R.id.tvID)
            tvName = itemView.findViewById(R.id.tvName)
            tvAge = itemView.findViewById(R.id.tvAge)
            tvBookNo = itemView.findViewById(R.id.tvBookNo)
        }
    }

    fun addDataToDataModel(models: List<ClientDataModel>) {
        this.modelList = models
        notifyDataSetChanged()
    }

    companion object {
        private val TAG = RecyclerViewAdapter::class.java.simpleName
    }
}
